[refactor] make m_pPlayer private and expose it through method GetAppPlayer()

xhaggi opened the pull request:
see commits

xhaggi committed:
[cleanup] no need to cast to int in CSeekHandler::SetSeekSize

xhaggi committed:
[cleanup] fix method scopes in CDataCacheCore

xhaggi committed:
[refactor] make m_pPlayer private and expose it through method GetAppPlayer()

FernetMenta commented on xbmc/Application.cpp:
```diff
@@ -4864,6 +4865,11 @@ std::string CApplication::GetCurrentPlayer()
 
+std::shared_ptr<CApplicationPlayer> CApplication::GetAppPlayer()
+{
+  return m_pPlayer;
```
m_pPlayer is a unique_ptr. not sure if passing it away as shared_ptr make sense.

FernetMenta commented on xbmc/GUIInfoManager.cpp:
```diff
@@ -6048,36 +6048,36 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
   case PLAYER_CHAPTERNAME:
-    if(g_application.m_pPlayer->IsPlaying())
-      g_application.m_pPlayer->GetChapterName(strLabel);
+    if(g_application.GetAppPlayer()->IsPlaying())
```
with regard on performance we should think about a better solution. now that's 2 function calls extra and 2 times a shared ptr get constructed/destructed.

FernetMenta commented on xbmc/GUIInfoManager.cpp:
```diff
@@ -6274,22 +6274,22 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
     break;
   case VIDEOPLAYER_SUBTITLES_LANG:
-    if(g_application.m_pPlayer && g_application.m_pPlayer->IsPlaying() && g_application.m_pPlayer->GetSubtitleVisible())
+    if(g_application.GetAppPlayer() && g_application.GetAppPlayer()->IsPlaying() && g_application.GetAppPlayer()->GetSubtitleVisible())
```
here it is even worse. you could get a ref to player and then use this for the following calls.

FernetMenta commented on xbmc/Application.h:
```diff
@@ -502,6 +501,7 @@ class CApplication : public CXBApplicationEx, public IPlayerCallback, public IMs
   std::atomic_uint m_WaitingExternalCalls;        /*!< counts threads wich are waiting to be processed in FrameMove */
   unsigned int m_ProcessedExternalCalls;          /*!< counts calls wich are processed during one "door open" cycle in FrameMove */
   unsigned int m_ProcessedExternalDecay = 0;      /*!< counts to close door after a few frames of no python activity */
+  std::shared_ptr<CApplicationPlayer> m_pPlayer;
```
now thinking more about it I'd say that a pointer makes no sense here. It never gets deleted and nothing checks the pointer for being valid. Better make it a normal member. CAppPlayer wraps the actual player and does the checks for nullptr.

xhaggi replied:
Ok will change it to be a normal member like we did for CSeekHandler

xhaggi replied:
done + force-pushed

xhaggi committed:
[refactor] rename m_pPlayer to m_appPlayer

FernetMenta commented on xbmc/GUIInfoManager.cpp:
```diff
@@ -6136,11 +6136,11 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
     break;
   case PLAYER_PLAYSPEED:
-      if(g_application.m_pPlayer->IsPlaying())
+      if(g_application.GetAppPlayer().IsPlaying())
```
all those condition checks are rather useless/wrong and can be dropped
- there is no guarantee that player is still playing when entering the block
- if player is not playing the methods return defaults
- unlikely that a skins query those labels if player is not playing

xhaggi replied:
done in a separate commit

FernetMenta commented on xbmc/GUIInfoManager.cpp:
```diff
@@ -6225,38 +6225,38 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
     break;
   case VIDEOPLAYER_AUDIO_CHANNELS:
-    if(g_application.m_pPlayer->IsPlaying())
+    if(g_application.GetAppPlayer().IsPlaying())
```
also not needed. m_audio/videoInfo can be reset to defaults, if player is not playing

xhaggi replied:
done

FernetMenta commented on xbmc/SeekHandler.cpp:
```diff
@@ -202,9 +202,9 @@ int CSeekHandler::GetSeekSize() const
-  int64_t playTime = g_application.m_pPlayer->GetTime();
-  double minSeekSize = (g_application.m_pPlayer->GetMinTime() - playTime) / 1000;
-  double maxSeekSize = (g_application.m_pPlayer->GetMaxTime() - playTime) / 1000;
+  int64_t playTime = g_application.GetAppPlayer().GetTime();
```
better requesting the ref only once

xhaggi committed:
[refactor] make m_stackHelper private and expose it through method GetAppStackHelper()

xhaggi commented:
@Voyager1 FYI .. I changed the unique ptr `CApplicationStackHelper` to be a plain member of `CApplication` and made it private with a public method `GetAppStackHelper`.

xhaggi commented:
jenkins build this please

xhaggi commented:
@FernetMenta would be nice if you could take another look so I don't need another rebase round.

FernetMenta commented on xbmc/GUIInfoManager.cpp:
```diff
@@ -7223,88 +7190,88 @@ bool CGUIInfoManager::GetBool(int condition1, int contextWindow, const CGUIListI
     bReturn = (slideShow && slideShow->GetCurrentSlide() && slideShow->GetCurrentSlide()->IsVideo());
   }
-  else if (g_application.m_pPlayer->IsPlaying())
+  else if (g_application.GetAppPlayer().IsPlaying())
```
why did you keep this?

xhaggi replied:
I'm not sure if that is fine to remove all `IsPlaying` check in `CGUIInfoManager `

xhaggi replied:
dropped all `IsPlaying` checks from GUIInfoManager .. runtime tested and it looks good.

xhaggi committed:
[refactor] use switch+case instead of if+elseif in CGUIInfoManager::GetBool

This reduces dupe code and is necessary to easily drop the IsPlaying condition in the middle of the previous elseif conditions.

xhaggi committed:
[refactor] drop superfluous IsPlaying checks from GUIInfoManager

FernetMenta commented:
+1

Voyager1 commented:
> I changed the unique ptr CApplicationStackHelper to be a plain member of CApplication and made it private with a public method GetAppStackHelper.

seems ok to me.

xhaggi commented:
Jenkins build this please
